swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -527,21 +528,33 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
   /**
    * Partitions the list of marked resources to process:
    * For convenience, partitions are grouped by relevance
+   *
+   * Sets the partition offset time based on the delete-spread-ms configuration divided by the total
+   * partition size.
    */
   fun partitionList(
     markedResources: List<MarkedResource>,
     configuration: WorkConfiguration
-  ): List<List<MarkedResource>> {
-    val partitions = mutableListOf<List<MarkedResource>>()
+  ): List<ResourcePartition> {
+    val partitions = mutableListOf<ResourcePartition>()
     markedResources.groupBy {
       it.resource.grouping
     }.map {
-      Lists.partition(it.value, configuration.itemsProcessedBatchSize).forEach { partition ->
-        partitions.add(partition)
+      Lists.partition(it.value, configuration.itemsProcessedBatchSize).forEach { groupedMarkedResources ->
+        partitions.add(ResourcePartition(markedResources = groupedMarkedResources))
       }
     }
 
-    return partitions.sortedByDescending { it.size }
+    if (partitions.isNotEmpty()) {
+      val offsetIntervalMs = configuration.getDeleteSpreadMs(dynamicConfigService) / partitions.size
+      var offsetMs = 0L
+      partitions.forEach { partition ->
```

Comment:
for some extra fancy pantsiness:

```suggestion
      partitions.forEachIndexed { index, partition ->
        partition.offsetMs = offsetMs + index * offsetIntervalMs
      }
```